web/src/main/java/se/inera/intyg/intygstjanst/web/service/repo/model/CitizenCertificateConverterImpl.java

```diff
@@ -0,0 +1,63 @@
+package se.inera.intyg.intygstjanst.web.service.repo.model;
+
+import org.springframework.stereotype.Service;
+import se.inera.intyg.common.support.model.CertificateState;
+import se.inera.intyg.intygstjanst.persistence.model.dao.Certificate;
+import se.inera.intyg.intygstjanst.persistence.model.dao.CertificateStateHistoryEntry;
+import se.inera.intyg.intygstjanst.persistence.model.dao.Relation;
+import se.inera.intyg.intygstjanst.web.service.dto.citizen.*;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class CitizenCertificateConverterImpl implements CitizenCertificateConverter {
+    private final CitizenCertificateRelationConverter citizenCertificateRelationConverter;
+
+    public CitizenCertificateConverterImpl(CitizenCertificateRelationConverter citizenCertificateRelationConverter) {
+        this.citizenCertificateRelationConverter = citizenCertificateRelationConverter;
+    }
+
+    @Override
+    public CitizenCertificate get(Certificate certificate, List<Relation> relations) {
+        final var sentState = getSentState(certificate.getStates());
+
+        return CitizenCertificate
+                .builder()
+                .id(certificate.getId())
+                .type(certificate.getType())
+                .typeVersion(certificate.getTypeVersion())
+                .additionalInfo(certificate.getAdditionalInfo())
+                .issuerName(certificate.getSigningDoctorName())
+                .unitId(certificate.getCareUnitId())
+                .unitName(certificate.getCareUnitName())
+                .issued(certificate.getSignedDate())
+                .sentDate(sentState.map(CertificateStateHistoryEntry::getTimestamp).orElse(null))
+                .relations(getRelations(certificate.getId(), relations))
+                .build();
+    }
+
+    private CitizenCertificateRelationDTO getRelation(String certificateId, Relation relation) {
+        return citizenCertificateRelationConverter.get(
```

Comment:
Use the whole relation instead of get:ing each value and calling the converter. Reduces number of arguments.